fastly/platform_tls.go

```diff
@@ -0,0 +1,390 @@
+package fastly
+
+import (
+	"fmt"
+	"time"
+
+	"github.com/google/jsonapi"
+)
+
+// GetPlatformPrivateKeysInput is an input to the GetPlatformPrivateKeys function.
+// Allowed values for the fields are described at https://docs.fastly.com/api/platform-tls
+type GetPlatformPrivateKeyInput struct {
+	ID string
+}
+
+// PlatformPrivateKeysResponse is a response from the service tls API endpoint
+type PlatformPrivateKeysResponse struct {
+	Status  string                `mapstructure:"status"`
+	Meta    map[string]string     `mapstructure:"meta"`
+	Message string                `mapstructure:"msg"`
+	Data    []*PlatformPrivateKey `mapstructure:"data"`
+}
+
+// PlatformPrivateKeyResponse is a response from the service tls API endpoint
+type PlatformPrivateKeyResponse struct {
+	Status  string              `mapstructure:"status"`
+	Meta    map[string]string   `mapstructure:"meta"`
+	Message string              `mapstructure:"msg"`
+	Data    *PlatformPrivateKey `mapstructure:"data"`
+}
+
+// PlatformPrivateKey .
+type PlatformPrivateKey struct {
+	ID            string                        `mapstructure:"id"`
+	Type          string                        `mapstructure:"type"`
+	Attributes    PlatformPrivateKeysAttributes `mapstructure:"attributes"`
+	Comment       string                        `mapstructure:"comment"`
+	CustomerID    string                        `mapstructure:"customer_id"`
+	CreatedAt     *time.Time                    `mapstructure:"created_at"`
+	UpdatedAt     *time.Time                    `mapstructure:"updated_at"`
+	DeletedAt     *time.Time                    `mapstructure:"deleted_at"`
+	ActiveVersion uint                          `mapstructure:"version"`
+	Versions      []*Version                    `mapstructure:"versions"`
+}
+
+// PlatformPrivateKeysAttributes .
+type PlatformPrivateKeysAttributes struct {
+	KeyLength     string     `mapstructure:"key_length"`
+	KeyType       string     `mapstructure:"key_type"`
+	Name          string     `mapstructure:"name"`
+	CreatedAt     *time.Time `mapstructure:"created_at"`
+	Replace       bool       `mapstructure:"replace"`
+	PublicKeySHA1 string     `mapstructure:"public_key_sha1"`
+}
```

Comment:
As this is a JSONAPI style API format, we can use the [`jsonapi`](https://github.com/google/jsonapi) package to help marshal/unmarshal JSONAPI responses. Therefore, you shouldn't need to model the response data as three separate structs like this as the information we really care about which models a "private key" is the attributes field of the response. Therefore, `PlatformPrivateKeyAttributes` should become `PlatformPrivateKey` and the rest should be removed. You will need to use `jsonapi:""` field tags instead of the `mapstructure` ones.